libraryValidations/Python/test_json_validations_with_provider.py

```diff
@@ -0,0 +1,124 @@
+        return False
+    return enabled
+
+
+class TestFromProvider(unittest.TestCase):
+    # method: is_enabled
+    def test_no_filters(self):
+        test_key = "ProviderTelemetry"
+        with patch("featuremanagement.azuremonitor._send_telemetry.azure_monitor_track_event") as mock_track_event:
+            self.run_tests(test_key, mock_track_event)
+
+    @staticmethod
+    def load_from_provider():
+        connection_string = os.getenv("APP_CONFIG_VALIDATION_CONNECTION_STRING")
+        config = load(connection_string=connection_string, selects=[], feature_flag_enabled=True)
+
+        feature_manager = FeatureManager(config, on_feature_evaluated=publish_telemetry)
+        assert feature_manager is not None
+
+        return feature_manager
+
+    # method: is_enabled
+    def run_tests(self, test_key, track_event_mock):
+        feature_manager = self.load_from_provider()
+
+        with open(FILE_PATH + test_key + TESTS_JSON_KEY, "r", encoding="utf-8") as feature_flag_test_file:
+            feature_flag_tests = json.load(feature_flag_test_file)
+
+        for feature_flag_test in feature_flag_tests:
+            is_enabled = feature_flag_test[IS_ENABLED_KEY]
+            get_variant = feature_flag_test.get(GET_VARIANT_KEY, None)
+            expected_is_enabled_result = convert_boolean_value(is_enabled.get(RESULT_KEY))
+            feature_flag_id = test_key + "." + feature_flag_test[FEATURE_FLAG_NAME_KEY]
+            telemetry = feature_flag_test.get("Telemetry", None)
+
+            failed_description = f"Test {feature_flag_id} failed. Description: {feature_flag_test[DESCRIPTION_KEY]}"
+
+            if isinstance(expected_is_enabled_result, bool):
+                user = feature_flag_test[INPUTS_KEY].get(USER_KEY, None)
+                groups = feature_flag_test[INPUTS_KEY].get(GROUPS_KEY, [])
+                assert (
+                    feature_manager.is_enabled(
+                        feature_flag_test[FEATURE_FLAG_NAME_KEY], TargetingContext(user_id=user, groups=groups)
+                    )
+                    == expected_is_enabled_result
+                ), failed_description
+            else:
+                with raises(ValueError) as ex_info:
+                    feature_manager.is_enabled(feature_flag_test[FEATURE_FLAG_NAME_KEY])
+                expected_message = is_enabled.get(EXCEPTION_KEY)
+                assert str(ex_info.value) == expected_message, failed_description
+
+            if get_variant is not None and get_variant[RESULT_KEY]:
+                user = feature_flag_test[INPUTS_KEY].get(USER_KEY, None)
+                groups = feature_flag_test[INPUTS_KEY].get(GROUPS_KEY, [])
+                variant = feature_manager.get_variant(feature_flag_test[FEATURE_FLAG_NAME_KEY], TargetingContext(user_id=user, groups=groups))
+                if not variant:
+                    logger.error(f"Variant is None for {feature_flag_id}")
+                    assert False, failed_description
```

Comment:
I did it this way originally because it was the only way to only print the log if there was an issue. I guess we are mostly expecting it to work so shouldn't be needed, but it was useful when setting up everything.

Thoughts?

Reply:
Then we should remove the logger right? the logging is done only once and it feels weird